src/Compiler/Compiler.cpp

```diff
@@ -1013,6 +1017,51 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
     interp->registerCmd("detailed_placement", placement, this, 0);
     interp->registerCmd("place", placement, this, 0);
 
+    auto pin_loc_assign_method = [](void* clientData, Tcl_Interp* interp,
+                                    int argc, const char* argv[]) -> int {
+      Compiler* compiler = (Compiler*)clientData;
+
+      auto setPlaceOption = [compiler](std::string arg) {
+        if (arg == "random") {
+          compiler->PinAssignOpts(Compiler::PinAssignOpt::Random);
+        } else if (arg == "in_define_order") {
+          compiler->PinAssignOpts(Compiler::PinAssignOpt::In_Define_Order);
+        } else {
+          compiler->ErrorMessage("Unknown Placement Option: " + arg);
+        }
+      };
+
+      // If we received a tcl argument
+      if (argc > 1) {
```

Comment:
What happened if user call _pin_loc_assign_method random in_define_order_ ?
I think we have to make sure we have 2 args and just take second

Reply:
It world return "Unknown command" but will set the first command. Let me check how to avoid this.

Reply:
it will take the second arg if  user call like you mentioned.

Reply:
you didn't fix this. you don't need **for** loop.

Reply:
for loop has removed with if else.